File: app/src/main/java/nl/fontys/smpt42_1/fontysswipe/MainActivity.java
package nl.fontys.smpt42_1.fontysswipe;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.ImageButton;

import com.daprlabs.aaron.swipedeck.SwipeDeck;

import nl.fontys.smpt42_1.fontysswipe.adapter.SwipeDeckAdapter;

/**
 * @author SMPT42-1
 */
public class MainActivity extends AppCompatActivity {

    private static final int SWIPE_ANIMATION_DURATION = 200;

    private SwipeDeck swipeDeck;
    private ImageButton imageButtonSwipeLeft;
    private ImageButton imageButtonSwipeRight;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        bindViews();
        loadActivity();
    }

    private void bindViews() {
        swipeDeck = (SwipeDeck) findViewById(R.id.swipe_deck);
        imageButtonSwipeLeft = (ImageButton) findViewById(R.id.image_button_swipe_left);
        imageButtonSwipeRight = (ImageButton) findViewById(R.id.image_button_swipe_right);
    }

    private void loadActivity() {
        SwipeDeckAdapter swipeDeckAdapter = new SwipeDeckAdapter(TestData.questions);
        swipeDeck.setAdapter(swipeDeckAdapter);
        swipeDeck.setLeftImage(R.id.left_image);
        swipeDeck.setRightImage(R.id.right_image);

        imageButtonSwipeLeft.setOnClickListener(imageButtonSwipeLeftOnClickListener);

        imageButtonSwipeRight.setOnClickListener(imageButtonSwipeRightOnClickListener);
    }

    private View.OnClickListener imageButtonSwipeLeftOnClickListener = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            swipeDeck.swipeTopCardLeft(SWIPE_ANIMATION_DURATION);
        }
    };

    private View.OnClickListener imageButtonSwipeRightOnClickListener = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            swipeDeck.swipeTopCardRight(SWIPE_ANIMATION_DURATION);
        }
    };

}

File: app/src/main/java/nl/fontys/smpt42_1/fontysswipe/MainActivity.kt
package nl.fontys.smpt42_1.fontysswipe

import android.content.Intent
import android.content.Intent.*
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View.INVISIBLE
import android.view.View.VISIBLE
import com.daprlabs.aaron.swipedeck.SwipeDeck
import kotlinx.android.synthetic.main.activity_main.*
import nl.fontys.smpt42_1.fontysswipe.adapter.SwipeDeckAdapter
import nl.fontys.smpt42_1.fontysswipe.controller.SwipeController
import nl.fontys.smpt42_1.fontysswipe.controller.SwipeControllerListener

/**
 * @author SMPT42-1
 */
class MainActivity : AppCompatActivity(), SwipeControllerListener {

    val controller: SwipeController = SwipeController.createInstance(this)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        showViewElements(false)
    }

    override fun onSwipeControllerInitialized() {
        swipe_deck.setAdapter(SwipeDeckAdapter(controller.questions))
        swipe_deck.setLeftImage(R.id.image_view_left)
        swipe_deck.setRightImage(R.id.image_view_right)
        swipe_deck.setCallback(object : SwipeDeck.SwipeDeckCallback {
            override fun cardSwipedLeft(stableId: Long) {
                controller.processQuestion(false); number_progress_bar.progress = getProgressPercentage()
            }

            override fun cardSwipedRight(stableId: Long) {
                controller.processQuestion(true); number_progress_bar.progress = getProgressPercentage()
            }
        })

        number_progress_bar.max = 100
        image_button_swipe_left.setOnClickListener({ swipe_deck.swipeTopCardLeft(0); number_progress_bar.progress = getProgressPercentage() })
        image_button_swipe_right.setOnClickListener({ swipe_deck.swipeTopCardRight(0); number_progress_bar.progress = getProgressPercentage() })

        showViewElements(true)
    }

    private fun getProgressPercentage(): Int {
        val totalQuestionCount = SwipeController.getInstance().totalQuestionsCount.toDouble()
        val currentQuestionCount = SwipeController.getInstance().currentQuestionCount.toDouble()
        val percentage = ((currentQuestionCount / totalQuestionCount) * 100).toInt()
        return percentage
    }

    override fun onAllQuestionsProcessed() {
        val intent = Intent(this, ResultActivity::class.java)
        intent.addFlags(FLAG_ACTIVITY_CLEAR_TOP or FLAG_ACTIVITY_CLEAR_TASK or FLAG_ACTIVITY_NEW_TASK)
        startActivity(intent)
    }

    private fun showViewElements(show: Boolean) {
        progress_bar.visibility = if (show) INVISIBLE else VISIBLE
        image_button_swipe_left.visibility = if (show) VISIBLE else INVISIBLE
        image_button_swipe_right.visibility = if (show) VISIBLE else INVISIBLE
        number_progress_bar.visibility = if (show) VISIBLE else INVISIBLE
    }

}
